src/Analyzer/FunctionNode.cpp

```diff
@@ -209,15 +209,19 @@ ASTPtr FunctionNode::toASTImpl(const ConvertToASTOptions & options) const
         function_ast->kind = ASTFunction::Kind::WINDOW_FUNCTION;
     }
 
+    auto new_options = options;
```

Comment:
Need to discuss this, this feature `add_cast_for_constants` was added for very specific case.